challenge_6.js

```diff
@@ -6,11 +6,26 @@
  */
 exports.SumOfASquare = SumOfASquare;
 
-function SumOfASquare(naturalNumbers){
+function SumOfASquare(naturalNumbers) {
   this.naturalNumbers = naturalNumbers;
-}
-
-// do work here
 
+  this.sumOfSquares = function(){
+    let allSquaredSum = 0;
+    for (let i=1; i<=this.naturalNumbers; i++){
+      allSquaredSum += Math.pow(i, 2);
+    }
+    return allSquaredSum
+  }
+  
+  this.squareOfTheSums = function(){
+    let allSumBeforeSquare = 0;
+    let allSumSquared = 0;
+    for (let j=1; j<=this.naturalNumbers;j++){
+      allSumBeforeSquare += j;
+      allSumSquared = Math.pow(allSumBeforeSquare, 2);
```

Comment:
This should be pulled out to after the for loop is completed. This does unnecessary calculations within the for loop when you just need the square after the sum has been calculated.